src/block/block_ckpt.c

```diff
@@ -63,16 +64,18 @@ __wt_block_checkpoint_load(WT_SESSION_IMPL *session, WT_BLOCK *block, const uint
         ci = &_ci;
         WT_ERR(__wt_block_ckpt_init(session, ci, "checkpoint"));
     } else {
-#ifdef HAVE_DIAGNOSTIC
         /*
          * We depend on the btree level for locking: things will go bad fast if we open the live
          * system in two handles, or salvage, truncate or verify the live/running file.
          */
         __wt_spin_lock(session, &block->live_lock);
-        WT_ASSERT(session, block->live_open == false);
+        live_open = block->live_open;
         block->live_open = true;
         __wt_spin_unlock(session, &block->live_lock);
-#endif
+        WT_ERR_ASSERT(session, live_open == false, EBUSY, "attempt to re-open live file");
+        if (live_open)
+            WT_ERR_MSG(session, WT_ERROR, "%s: already open", block->name);
```

Comment:
I'm not sure of the intent of the two error checks here. Both are triggered if `live_open == true`, but we'll never get to the second. If compiled with `HAVE_DIAGNOSTIC`, `WT_ERR_ASSERT` will abort. Otherwise it calls `WT_ERR_MSG` which includes a `goto err` statement.

Reply:
Good catch.  I'll remove the second of these, I like keeping the assert  in diagnostic mode.

Reply:
I think it would also be useful to capture `block->name" from the second message.  But no reason it can't be part of the 1st message.